Embedding with IPEX optimize fails because the model is not in eval mode yet

## Describe the bug

Testing with IPEX library and IPEX_OPTMIZE enabled found that the model must be in eval() mode.

EmbeddingModule._optimize(model, ipex, device)\n  File \"/opt/app-root/lib64/python3.9/site-packages/caikit_nlp/modules/text_embedding/embedding.py\", line 225, in _optimize\n    model = ipex.optimize(model)\n  File \"/opt/app-root/lib64/python3.9/site-packages/intel_extension_for_pytorch/frontend.py\", line 337, in optimize\n    assert optimizer is not None, \"The optimizer should be given for training mode\"\nAssertionError: The optimizer should be given for training mode", "level": "error", "log_code": "<RUN62912924E>", "message": "load failed when processing path: demo/models/intfloat/multilingual-e5-large with error: AssertionError('The optimizer should be given for training mode')", "model_id": "intfloat/multilingual-e5-large", "num_indent": 0, "thread_id": 140054211044928, "timestamp": "2024-02-06T20:09:18.159280"}

--- a/caikit_nlp/modules/text_embedding/embedding.py
+++ b/caikit_nlp/modules/text_embedding/embedding.py
@@ -153,6 +153,7 @@ class EmbeddingModule(ModuleBase):
         ipex = cls._get_ipex()
         device = cls._select_device(ipex)
         model = SentenceTransformer(model_name_or_path=artifacts_path, device=device)
+        model.eval()  # required for IPEX at least
         if device is not None:
             model.to(torch.device(device))
         model = EmbeddingModule._optimize(model, ipex, device)

